### PR TITLE
Improve routing

### PR DESCRIPTION
**Pull request by hlohaus**
### 1. Why is this change necessary?

1. Decorating the router is not necessary anymore.
2. We don't have to clone the request.
3. Admin now use basePath for static urls. eg. /index.php/admin works now.
4. Adds "IS_CANONICAL_SEO_URL_REQUEST" attribute. It is set, if it is a seo url and canonical. (i need that for my plugin)
4. Fix baseUrl and basePath confusions

FYI:

basePath = /public
baseUrl = /public/index.php
or (now)
baseUrl = /public/de

### 2. What does this change do, exactly?

see code

### 3. Describe each step to reproduce the issue or behaviour.

Try to open the shop with /public/index.php.

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Storefront/Framework/Routing/Router.php [hunk 1]
```diff
@@ -14,9 +13,6 @@
 
 class Router implements RouterInterface, RequestMatcherInterface, WarmableInterface, ServiceSubscriberInterface
```
Comment (mitelg) [inline src/Storefront/Framework/Routing/Router.php]: this class only calls the decorated one. so it is obsolete?

Comment (hlohaus) [reply src/Storefront/Framework/Routing/Router.php]: yes it is, maybe for second pull request

**Comment by OliverSkroblin** [comment]
Hi @hlohaus,

Thanks for your contribution. I imported your pull request. Currently more than 50 tests fail, including the `\Shopware\Storefront\Test\Framework\Routing\RouterTest` which checks the domain routing. 

```

Failed asserting that two strings are identical.
Expected :'http://test.de:8000/de/'
Actual   :'http://test.de:8000/'
 
Failed asserting that two strings are identical.
Expected :'http://test.de/de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
Actual   :'http://test.de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
 
Failed asserting that two strings are identical.
Expected :'http://test.de:8000/de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
Actual   :'http://test.de:8000/navigation/fca68da7b53444ac89fc51dd55c4ede4'
 
Failed asserting that two strings are identical.
Expected :'de/'
Actual   :''
 
Failed asserting that two strings are identical.
Expected :'/de/'
Actual   :'/'
 
Failed asserting that two strings are identical.
Expected :'de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
Actual   :'navigation/fca68da7b53444ac89fc51dd55c4ede4'
 
Failed asserting that two strings are identical.
Expected :'/de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
Actual   :'/navigation/fca68da7b53444ac89fc51dd55c4ede4'
 
Failed asserting that two strings are identical.
Expected :'http://test.de:8000/de/'
Actual   :'http://test.de:8000/'
 
Failed asserting that two strings are identical.
Expected :'de/'
Actual   :''
 
Failed asserting that two strings are identical.
Expected :'//test.de:8000/de/'
Actual   :'//test.de:8000/'
 
Failed asserting that two strings are identical.
Expected :'//test.de/de/'
Actual   :'//test.de/'
 
Failed asserting that two strings are identical.
Expected :'//test.de:8000/de/'
Actual   :'//test.de:8000/'
 
Failed asserting that two strings are identical.
Expected :'de/'
Actual   :''
 
Failed asserting that two strings are identical.
Expected :'//test.de:8000/de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
Actual   :'//test.de:8000/navigation/fca68da7b53444ac89fc51dd55c4ede4'
 
Failed asserting that two strings are identical.
Expected :'//test.de:8000/de/'
Actual   :'//test.de:8000/'
 
Failed asserting that two strings are identical.
Expected :'http://test.de/de/'
Actual   :'http://test.de/'
 
Failed asserting that two strings are identical.
Expected :'http://test.de:8000/de/'
Actual   :'http://test.de:8000/'
 
Failed asserting that two strings are identical.
Expected :'/de/'
Actual   :'/'
 
Failed asserting that two strings are identical.
Expected :'/de/'
Actual   :'/'
 
Failed asserting that two strings are identical.
Expected :'//test.de/de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
Actual   :'//test.de/navigation/fca68da7b53444ac89fc51dd55c4ede4'
 
Failed asserting that two strings are identical.
Expected :'http://test.de/de/'
Actual   :'http://test.de/'
 
Failed asserting that two strings are identical.
Expected :'//test.de/de/'
Actual   :'//test.de/'
 
Failed asserting that two strings are identical.
Expected :'http://test.de/de/'
Actual   :'http://test.de/'
 
Failed asserting that two strings are identical.
Expected :'//test.de/de/'
Actual   :'//test.de/'
 
```

**Comment by OliverSkroblin** [comment]
To execute the tests simply use the ./psh.phar init & ./psh.phar unit command.

**Comment by OliverSkroblin** [comment]
Deleting a unit test is not fixing a unit test

**Comment by OliverSkroblin** [comment]
Sorry, my mistake.

Git shows me that you deleted the `src/Storefront/Test/Framework/Routing/RouterTest.php`  ;)

**Comment by hlohaus** [comment]
Yes, i fixed it. And src/Storefront/Test/Framework/Routing/StorefrontRoutingTest.php now too.

**Comment by OliverSkroblin** [comment]
Hi @hlohaus,

Thanks for your contribution. We tested your changes. Several bugs have been encountered. Since we are just about to release the next version, we want to avoid doing a complete refactoring of the sales channel routing.

**Pull request closed without merging**